src/rest-server/src/models/v2/job/k8s.js

```diff
@@ -456,7 +472,30 @@ const generateTaskRole = (frameworkName, taskRole, labels, config, storageConfig
                   ...infinibandDevice && {'rdma/hca': 1},
                 },
               },
-              env: [],
+              env: [
+                ...frameworkEnvList,
+                {
+                  name: 'PAI_CURRENT_TASK_ROLE_NAME',
+                  value: taskRole,
+                },
+                {
+                  name: 'PAI_CURRENT_TASK_ROLE_CURRENT_TASK_INDEX',
+                  valueFrom: {
+                    fieldRef: {
+                      fieldPath: `metadata.annotations['FC_TASK_INDEX']`,
+                    },
+                  },
+                },
+                // backward compatibility
+                {
+                  name: 'PAI_TASK_INDEX',
+                  valueFrom: {
+                    fieldRef: {
+                      fieldPath: `metadata.annotations['FC_TASK_INDEX']`,
+                    },
+                  },
+                },
```

Comment:
create `taskRoleEnvList` in `generateTaskRole` and pass to both init container and job container?